components/brave_ads/browser/ads_service_factory.cc

```diff
@@ -76,10 +80,41 @@ bool AdsServiceFactory::ServiceIsNULLWhileTesting() const {
 
 void AdsServiceFactory::RegisterProfilePrefs(
     user_prefs::PrefRegistrySyncable* registry) {
+  if (ShouldMigratePrefs(registry)) {
+    // prefs::kBraveAdsPrefsVersion should default to 1 for legacy installations
+    // so that preferences are migrated from version 1 to the current version
+    registry->RegisterIntegerPref(prefs::kBraveAdsPrefsVersion,
+        prefs::kBraveAdsPrefsDefaultVersion);
+  } else {
+    registry->RegisterIntegerPref(prefs::kBraveAdsPrefsVersion,
+        prefs::kBraveAdsPrefsCurrentVersion);
+  }
+
   registry->RegisterBooleanPref(prefs::kBraveAdsEnabled, false);
+
   registry->RegisterUint64Pref(prefs::kBraveAdsPerHour, 2);
-  registry->RegisterUint64Pref(prefs::kBraveAdsPerDay, 6);
+
+  #if defined(OS_ANDROID)
+    registry->RegisterUint64Pref(prefs::kBraveAdsPerDay, 12);
```

Comment:
Seems like we perform this particular check in at least two places, it might be good to make it a utility function or just define a platform-specific constant.

Reply:
This is the defaults for fresh installs, the other location is an upgrade path from mapping version 1 to 2.